extension/packagemanagerextension/nbextension/api.js

```diff
@@ -0,0 +1,117 @@
+define([
+    'jquery',
+    './urls'
+], function ($, urls) {
+    "use strict";
+
+    let endpoints = {
+        "update_packages": {
+            "uri": "packages",
+            "method": "PATCH"
+        },
+        "checkupdate": {
+            "uri": "packages/check_update?project=",
+            "method": "GET"
+        },
+        "delete_packages": {
+            "uri": "packages",
+            "method": "DELETE"
+        },
+        "install_packages": {
+            "uri": "packages",
+            "method": "POST"
+        },
+        "getinfo": {
+            "uri": "project_info?project=",
+            "method": "GET"
+        },
+        "search": {
+            "uri": "packages/search?q=",
+            "method": "GET"
+        }
+    };
+
+    /*
+    This function makes the API calls to the specified endpoint with the corresponding headers
+    */
+    function api_call(url, method, payload = "") {
+        let settings = {
+            "url": url,
+            "method": method,
+            "headers": {
+                "Content-Type": "application/json",
+                "cache-control": "no-cache",
+            },
+            "processData": false,
+            "data": payload
+        };
+
+        return new Promise(resolve => {
```

Comment:
You still haven't explain me why you need to return a promise...

Reply:
I'll be needing it for async/await.